nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1457,4 +1466,85 @@ public String getProcessGroupIdentifier() {
         final ProcessGroup group = getProcessGroup();
         return group == null ? null : group.getIdentifier();
     }
+
+    private static final class ProcessorDetails {
```

Comment:
Would recommend that this be externalized into its own class file, since this class already is at nearly 1500 lines of code, and this inner class is static. Not a big deal but worth considering.